.github/workflows/docker.yml

```diff
@@ -16,22 +22,33 @@ jobs:
         with:
           ruby-version: 2.6.x
       - run: gem install bundler
-      - name: Lint container
-        working-directory: docker
-        run: |
-          make lint
       - name: Build container
-        env:
-          PUPPERWARE_ANALYTICS_STREAM: production
-          IS_LATEST: true
         working-directory: docker
-        run: make build test
+        run: make lint build test
       - name: Publish container
-        env:
-          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
-          DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
-          IS_LATEST: true
         working-directory: docker
         run: |
           docker login -u "$DOCKERHUB_USERNAME" -p "$DOCKERHUB_PASSWORD"
           make publish
+      - name: Build release container
+        env:
+          IS_RELEASE: true
+        working-directory: docker
+        run: |
+          make prep
+          if [ $? -eq 0 ]; then
+            make lint build test
+          else
+            echo "Skipping release container building and testing"
+          fi
+      - name: Publish release container
+        env:
+          IS_RELEASE: true
+        working-directory: docker
+        run: |
+          make prep
+          if [ $? -eq 0 ]; then
+            make publish
+          else
+            echo "Skipping release container publishing"
```

Comment:
Do we need to make an exception for the `edge` build? We always publish that over existing `edge` builds, right?

Reply:
nope, the release container won't be tagged edge, so after a release is published we'll just have a container from source published and a container from packages published during the same gh action